src/libraries/System.Private.CoreLib/src/System/Threading/Thread.OSX.cs

```diff
@@ -0,0 +1,47 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+
+namespace System.Threading
+{
+    internal static class ThreadOSX
```

Comment:
>I assume that you have done this intentionally to avoid adding static constructor to Thread.

Yes.

>Or maybe this can be just a part of ObjectiveCMarshal and be under FEATURE_OBJCMARSHAL ifdef. It is all one bundle of Objective-C supporting features that we are very likely to be enabled individually.

I am all for doing this after we merge together the two features. At present the `FEATURE_OBJCMARSHAL` is introduced in https://github.com/dotnet/runtime/pull/52146 which means until that goes in I can't reconcile that. Once this goes in, I will merge main into that branch and put all of it under that feature build define.

Reply:
You give it the right names in this PR, and then the merge into the other branch will have smaller delta. I think it is better to avoid redundant renames in history, it makes it easier to see what happened. I do not care strong enough.

Reply:
Fair point. Okay.